doc/MAPGEN.md

```diff
@@ -1054,15 +1054,22 @@ For example, the default value of a parameter, or a terrain id in the
 * A JSON object containing the key `"param"`, whose corresponding value is the
   string name of a parameter as discussed in [Mapgen
   parameters](#mapgen-parameters).  For example, `{ "param": "roof_type" }`.
+  You may be required to also supply a fallback value, such as `{ "param":
+  "roof_type", "fallback": "t_flat_roof" }`.  The fallback is necessary to
+  allow mapgen definitions to change without breaking an ongoing game.
+  Different parts of the same overmap special can be generated at different
+  times, and if a new parameter is added to the definition part way through the
+  generation then the value of that parameter will be missing and the fallback
+  will be used.
```

Comment:
Correct.  Where "the start" is the time when the first OMT for the overmap special is mapgened.

(Also, any overmap which was generated by the code before this PR is merged won't get args for any of its specials, so all will use the fallbacks, but that's a short term problem).

You've reminded me, however, that I haven't implemented serialization for the new overmap data.  I'd better go do that.

Reply:
By the below, the fallback can be a distribution? I am unclear on the difference between using that and using the non-fallback distribution.

Reply:
No, currently the fallback has to be a value.  If it were a distribution then it would be sampled from for each tile, or maybe (with additional work) for each OMT, neither of which would seem like desirable behaviour.

Reply:
I am confused by the below "inconsistent default value distributions" debug message, then.

Reply:
A default is not the same as a fallback.  At the moment a 'default' is better just thought of as the value of a parameter.  But the long-term plan is that you can pass arguments to mapgen which will override the default, hence the term 'default'.  The distributions for default values have to match.  For example, two OMTs of the same overmap special can define an overmap-special-scoped parameter with the same name but different default values, and then mapgen won't know which one to use.